sklearn/cross_decomposition/pls_.py

```diff
@@ -820,7 +829,20 @@ def fit(self, X, Y):
         return self
 
     def transform(self, X, Y=None):
-        """Apply the dimension reduction learned on the train data."""
+        """
+        Apply the dimension reduction learned on the train data.
+
+        Parameters
+        ----------
+        X : array-like of predictors, shape = [n_samples, p]
+            Training vectors, where n_samples in the number of samples and
+            p is the number of predictors.
```

Comment:
Please be consistent between `fit`, `fit_transform`, and `transform`.
In particular, I would rather avoid using `p` and `q`, and fix them in `fit_transform` and in the attributes docstring